bears/general/FilenameBear.py

```diff
@@ -74,12 +83,44 @@ def run(self, filename, file,
         if ignore_uppercase_filenames and filename_without_extension.isupper():
             return
 
-        if new_name != filename_without_extension:
-            diff = Diff(file, rename=os.path.join(head, new_name + extension))
+        if (new_name != filename_without_extension or
+            not filename_without_extension.startswith(filename_prefix) or
+                not filename_without_extension.endswith(filename_suffix)):
+            correct_prefix_name = (new_name
+                                   if
+                                   filename_without_extension.startswith(
+                                                               filename_prefix)
+                                   else
+                                   filename_prefix + new_name)
+
+            correct_prefixsuffix_name = (correct_prefix_name
+                                         if
+                                         filename_without_extension.endswith(
+                                                               filename_suffix)
+                                         else
+                                         correct_prefix_name + filename_suffix)
+
+            diff = Diff(file, rename=os.path.join(
+                head, correct_prefixsuffix_name + extension))
+
+            FilenameBear_messages = [
```

Comment:
why did you prefix with `FilenameBear`? I think just `messages` is totally fine :)